sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -118,7 +120,27 @@ public static PipelineOptions createPipelineOptions(Map<String, String> map) {
     for (Map.Entry<String, String> entry : map.entrySet()) {
       args[i++] = "--" + entry.getKey() + "=" + entry.getValue();
     }
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+
+    PipelineOptions options;
+    try {
+      options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+    } catch (PipelineOptionUnexpectedPropertyException e) {
+      String errorMessage =
+          String.format(
+              "Pipeline option '%s' is not supported in this context. It must be "
+                  + "cleared by 'RESET %s'.",
+              e.getPropertyName(), e.getPropertyName());
+
+      if (e.getClosestMatches().size() == 1) {
+        errorMessage +=
+            String.format(
+                " Did you mean 'SET %s'?", Iterables.getOnlyElement(e.getClosestMatches()));
+      } else if (e.getClosestMatches().size() > 1) {
+        errorMessage += String.format(" Did you mean SET one of %s?", e.getClosestMatches());
+      }
+      throw new IllegalArgumentException(errorMessage);
```

Comment:
Can you pass along the original exception as well? `throw new IllegalArgumentException(errorMessage, e)`

Reply:
+1

Reply:
Done!